src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            Sys.ActorOf(Props.Create(() => new Subject()), "subject");
+            EnterBarrier("subject-started");
+
+            RunOn(() =>
+            {
+                var secondAddress = Node(_specConfig.Second).Address;
+
+                Identify(_specConfig.Second, "subject");
+
+                EventFilter.Warning(new Regex("address is now gated")).ExpectOne(() =>
+                {
+                    RARP.For(Sys).Provider.Transport.ManagementCommand(
+                        new ForceDisassociateExplicitly(Node(_specConfig.Second).Address, DisassociateInfo.Unknown))
+                        .Wait(TimeSpan.FromSeconds(3));
+                });
+
+
+                EnterBarrier("gated");
+                TestConductor.Shutdown(_specConfig.Second).Wait();
+                Within(TimeSpan.FromSeconds(10), () =>
+                    {
+                        AwaitAssert(
+                            () =>
+                                {
+                                    Sys.ActorSelection(new RootActorPath(secondAddress) / "user" / "subject")
+                                        .Tell(new Identify("subject"));
+                                    ExpectMsg<ActorIdentity>();
+                                });
+                    });
+                Sys.ActorSelection(new RootActorPath(secondAddress)/"user"/"subject").Tell("shutdown");
+            }, _specConfig.First);
+
+            RunOn(() =>
+            {
+                var addr = Sys.AsInstanceOf<ExtendedActorSystem>().Provider.DefaultAddress;
+                var firstAddress = Node(_specConfig.First).Address;
+
+                EnterBarrier("gated");
+
+                Sys.WhenTerminated.Wait(TimeSpan.FromSeconds(10));
+
+                var sb = new StringBuilder();
+                sb.AppendLine("akka.remote.retry-gate-closed-for = 0.5s")
+                    .AppendLine(" akka.remote.helios.tcp {")
+                    .AppendLine("hostname = " + addr.Host)
+                    .AppendLine("port =" + addr.Port)
+                    .AppendLine("}");
+
+                var freshSystem = ActorSystem.Create(Sys.Name, ConfigurationFactory.ParseString(sb.ToString())
+                    .WithFallback(Sys.Settings.Config));
+
+                var probe = CreateTestProbe(freshSystem);
+
+                // Pierce the gate
+                Within(TimeSpan.FromSeconds(30), () =>
+                {
+                    AwaitAssert(() =>
+
+                    {
+                        freshSystem.ActorSelection(new RootActorPath(firstAddress)/"user"/"subject")
+                            .Tell(new Identify("subject"), probe);
+                        probe.ExpectMsg<ActorIdentity>();
+                    });
+                });
+
+                // Now the other system will be able to pass, too
+                freshSystem.ActorOf(Props.Create(() => new Subject()), "subject");
```

Comment:
Setting a breakpoint here allows the spec to pass... Interesting. I'll keep looking into it.